server/core/src/indexer/indexer.ts

```diff
@@ -293,6 +293,11 @@ export class FullTextIndexPipeline implements FullTextPipeline {
       this.indexesCreated = true
       // We need to be sure we have individual indexes per stage.
       for (const st of this.stages) {
+        await this.storage.removeOldIndex(
+          DOMAIN_DOC_INDEX_STATE,
```

Comment:
Let's enhance stage with clean pattern(regexp) and stageId, so it will clean all passed and keep required for every stage. it will be more clean.